payment/payment.go

```diff
@@ -37,19 +38,20 @@ func FromPublicKey(pubkey *btcec.PublicKey, net *network.Network) *Payment {
 		tmpNet = net
 	}
 	publicKeyBytes := pubkey.SerializeCompressed()
-	hash := hash160(publicKeyBytes)[:ripemd160.Size]
+	pkHash := hash160(publicKeyBytes)[:ripemd160.Size]
 	script := make([]byte, 0)
-	script = append([]byte{Op0, byte(len(hash))}, hash...)
-	return &Payment{tmpNet, pubkey, hash, nil, nil, script}
+	script = append([]byte{Op0, byte(len(pkHash))}, pkHash...)
+	witnessHash := sha256.Sum256(script)
+	return &Payment{tmpNet, pubkey, pkHash, nil, nil, script, witnessHash[:]}
```

Comment:
If I recall correctly the witness program must be 20bytes long, so would be better to bound the size of the hash

Reply:
@altafan

Reply:
witnessProgram is 20bytes for P2WPKH and 32bytes for P2WSH